### ISSUE TITLE
`ManagerRegistry#getRepository()` does not use `getManagerForClass()` internally

### ISSUE THREAD

**Issue by mnapoli** [doctrine/persistence#45]
Assuming the `Foo` class is not in the default entity manager:

- works:

```php
$objectManager = $managerRegistry->getManagerForClass(Foo::class);
$repository = $objectManager->getRepository(Foo::class);
```

- works:

```php
$repository = $managerRegistry->getRepository(Foo::class, <manager name>);
```

- doesn't work (because `Foo` is looked into the default manager):

```php
$repository = $managerRegistry->getRepository(Foo::class);
```

I would have thought the 3rd case would work: after all the manager registry can retrieve the correct manager using `getManagerForClass()` internally.

---

Would it make sense to have `getRepository()` use `getManagerForClass()` internally to retrieve the repository in the correct manager?

**Comment by alcaeus** [comment]
That is correct. After taking a look, I would suggest the following changes:
* If a manager name was given as second argument, it will always attempt to get that manager
* If no manager name was given, it will use `getManagerForClass` to get the first manager that handles this class
* If the previous steps did not yield an object manager, the default manager of the registry will be used.

I have the changes made but still need to add a bunch of tests around this functionality since `getRepository` wasn't tested at all yet.

**Comment by mnapoli** [comment]
oh awesome thank you @alcaeus!

**Comment by alcaeus** [comment]
This will be fixed in 1.1.1.

**Comment by mnapoli** [comment]
Thank you for working on this!
